python/plugins/processing/algs/qgis/PointOnSurface.py

```diff
@@ -66,26 +67,24 @@ def displayName(self):
         return self.tr('Point on surface')
 
     def processAlgorithm(self, parameters, context, feedback):
-        layer = QgsProcessingUtils.mapLayerFromString(self.getParameterValue(self.INPUT_LAYER), context)
+        source = self.parameterAsSource(self.INPUT_LAYER, context)
 
-        writer = self.getOutputFromName(
-            self.OUTPUT_LAYER).getVectorWriter(layer.fields(), QgsWkbTypes.Point, layer.crs(), context)
+        (sink, dest_id) = self.parameterAsSink(parameters, self.OUTPUT_LAYER, context, source.fields(), QgsWkbTypes.Point, source.crs())
 
-        features = QgsProcessingUtils.getFeatures(layer, context)
-        total = 100.0 / layer.featureCount() if layer.featureCount() else 0
+        features = source.getFeatures()
+        total = 100.0 / source.featureCount() if source.featureCount() else 0
 
         for current, input_feature in enumerate(features):
+            if feedback.isCancelled():
+                break
+
             output_feature = input_feature
             input_geometry = input_feature.geometry()
             if input_geometry:
                 output_geometry = input_geometry.pointOnSurface()
-                if not output_geometry:
-                    raise GeoAlgorithmExecutionException(
-                        self.tr('Error calculating point on surface'))
```

Comment:
I'm not sure it's not actually handling null input (the earlier check does that) and in which case I'd agree with you. Here's it's checking for null output from the GEOS method. I guess it depends on whether GEOS will ever error out when calculating the point on surface. I'd probably just leave this here and change it to QgsProcessingException, OR change it to `feedback.reportError(...)` so it doesn't break the algorithm's execution.